### DISCUSSION TITLE
block wrapper class is generated from block name, save to use?

### DISCUSSION THREAD

**Discussion by ivanjeremic** [WordPress/gutenberg#38074] [Q&A] [answered]
the wrapper class is currently generated by the name defined in block.json
`"name": "create-block/my-block",` becomes `<div class="create-block-my-block">..</div>`

I want to use that class inside my code for some important task, is that class generated on built time or on runtime? Is it save to be used or is there any chance that future updates change the way the class is generated and my code breaks?

**Comment by gziolo** [comment, marked as answer]
> Is it save to be used or is there any chance that future updates change the way the class is generated and my code breaks?

You can assume that his class will always stay the same for the given block `name`. 

There is one remark here. It's something that plugins can modify with a [`blocks.getBlockDefaultClassName`](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/filters/block-filters.md#blocksgetblockdefaultclassname) filter, but as far as I know this filter is used to inject more class names rather than change the underlying mechanism.
